wrapper/src/test/java/software/amazon/jdbc/DialectDetectionTests.java

```diff
@@ -0,0 +1,269 @@
+
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.ResultSetMetaData;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.Properties;
+import java.util.stream.Stream;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+import software.amazon.jdbc.dialect.AuroraMysqlDialect;
+import software.amazon.jdbc.dialect.AuroraPgDialect;
+import software.amazon.jdbc.dialect.Dialect;
+import software.amazon.jdbc.dialect.DialectManager;
+import software.amazon.jdbc.dialect.MariaDbDialect;
+import software.amazon.jdbc.dialect.MysqlDialect;
+import software.amazon.jdbc.dialect.PgDialect;
+import software.amazon.jdbc.dialect.RdsMultiAzDbClusterMysqlDialect;
+import software.amazon.jdbc.dialect.RdsMultiAzDbClusterPgDialect;
+import software.amazon.jdbc.dialect.RdsMysqlDialect;
+import software.amazon.jdbc.dialect.RdsPgDialect;
+import software.amazon.jdbc.exceptions.ExceptionManager;
+import software.amazon.jdbc.targetdriverdialect.TargetDriverDialect;
+
+public class DialectDetectionTests {
+  private static final String LOCALHOST = "localhost";
+  private static final String RDS_DATABASE = "database-1.xyz.us-east-2.rds.amazonaws.com";
+  private static final String RDS_AURORA_DATABASE = "database-2.cluster-xyz.us-east-2.rds.amazonaws.com";
+  private static final String MYSQL_PROTOCOL = "jdbc:mysql://";
+  private static final String PG_PROTOCOL = "jdbc:postgresql://";
+  private static final String MARIA_PROTOCOL = "jdbc:mariadb://";
+  @Mock private Connection mockConnection;
+  @Mock private Statement mockStatement;
+  @Mock private ResultSet successResultSet;
+  @Mock private ResultSet failResultSet;
+  @Mock private HostSpec mockHost;
+  @Mock private ConnectionPluginManager pluginManager;
+  @Mock private TargetDriverDialect mockTargetDriverDialect;
+  @Mock private ResultSetMetaData mockResultSetMetaData;
+  private final DialectManager dialectManager = new DialectManager(null);
+  private final Properties props = new Properties();
+  private AutoCloseable closeable;
+
+  @BeforeEach
+  void setUp() throws SQLException {
+    System.out.println("running set up");
+    closeable = MockitoAnnotations.openMocks(this);
+    when(this.mockConnection.createStatement()).thenReturn(this.mockStatement);
+    when(this.mockHost.getUrl()).thenReturn("url");
+    when(this.failResultSet.next()).thenReturn(false);
+    when(mockConnection.createStatement()).thenReturn(mockStatement);
+  }
+
+  @AfterEach
+  void cleanUp() throws Exception {
+    System.out.println("running clean up");
+    closeable.close();
+    Mockito.reset(
```

Comment:
It is a little bit confusing to see the code calling reset after closing the mocks. I thought close should be suffice?